File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
import java.util.*;
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        for(int i=0; i<n; i++){
             // 1. 이진수 변환 + 자리수 n 맞추기 (앞에 0 채움)
            String str1 = String.format("%" + n + "s", Integer.toBinaryString(arr1[i])).replace(' ', '0');
            String str2 = String.format("%" + n + "s", Integer.toBinaryString(arr2[i])).replace(' ', '0');

            // 2. 한 줄 만들기
            StringBuilder line = new StringBuilder();
            for (int j = 0; j < n; j++) {
                if (str1.charAt(j) == '1' || str2.charAt(j) == '1') {
                    line.append('#');
                } else {
                    line.append(' ');
                }
            }
            answer[i] = line.toString();
        }
        return answer;
    }
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    bin1 = []
    bin2=[]
    for i in arr1:
        row = list(bin(i)[2:].zfill(n).replace("1","#").replace("0"," "))
        bin1.append(row)
    for j in arr2:
        row = list(bin(j)[2:].zfill(n).replace("1","#").replace("0"," "))
        bin2.append(row)
        
    for i in range(n):
        row = ''
        for j in range(n):
            if bin1[i][j] == "#" or bin2[i][j] == "#":
                row+="#"
            else:
                row+=" "
        answer.append(row)
            
    return answer